File: src/main/java/com/adinsight/agent/tools/ToolRegistry.java
package com.adinsight.agent.tools;

import com.adinsight.agent.session.AgentSession;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class ToolRegistry {
    private final List<AgentTool<?>> tools;

    public ToolRegistry(List<AgentTool<?>> tools) {
        this.tools = List.copyOf(tools);
    }

    public Map<String, AgentTool<?>> getTools(AgentSession session) {
        Map<String, AgentTool<?>> enabledTools = new LinkedHashMap<>();
        for (AgentTool<?> tool : tools) {
            if (tool.isEnabled(session)) {
                enabledTools.put(tool.name(), tool);
            }
        }
        return enabledTools;
    }
}

File: src/tools/tool-registry.ts
import type { AgentSession } from "../session/agent-session.js";
import type { AgentTool } from "./agent-tool.js";

export class ToolRegistry {
  private readonly tools: readonly AgentTool[];

  constructor(tools: readonly AgentTool[]) {
    this.tools = [...tools];
  }

  getTools(session: AgentSession): ReadonlyMap<string, AgentTool> {
    const enabledTools = new Map<string, AgentTool>();

    for (const tool of this.tools) {
      if (tool.isEnabled == null || tool.isEnabled(session)) {
        enabledTools.set(tool.name, tool);
      }
    }

    return enabledTools;
  }
}
